crates/rust-analyzer/tests/heavy_tests/main.rs

```diff
@@ -676,3 +680,159 @@ pub fn foo(_input: TokenStream) -> TokenStream {
     let value = res.get("contents").unwrap().get("value").unwrap().to_string();
     assert_eq!(value, r#""```rust\nfoo::Bar\n```\n\n```rust\nfn bar()\n```""#)
 }
+
+#[test]
+fn test_issue_5103() {
```

Comment:
We don't add regression slow tests, as they are slow. So, this either has to be untested, or we need to move the checks to the type level.